lms/static/scripts/frontend_apps/utils/dashboard/hooks.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useMemo } from 'preact/hooks';
+import { useLocation, useSearch } from 'wouter-preact';
+
+import { queryStringToRecord, recordToQueryString } from '../url';
+
+export type DashboardFilters = {
+  studentIds: string[];
+  assignmentIds: string[];
+  courseIds: string[];
+};
+
+export type UseDashboardQueryFilters = {
+  filters: DashboardFilters;
+  updateFilters: (filtersToUpdate: Partial<DashboardFilters>) => void;
+};
+
+/**
+ * Allows to track dashboard filters as a piece of state synced in the query string
+ */
+export function useDashboardQueryFilters(): UseDashboardQueryFilters {
+  const search = useSearch();
+  const queryParams = useMemo(() => queryStringToRecord(search), [search]);
+  const filters = useMemo(() => {
+    const { student_id = [], course_id = [], assignment_id = [] } = queryParams;
+    const courseIds = Array.isArray(course_id) ? course_id : [course_id];
+    const assignmentIds = Array.isArray(assignment_id)
+      ? assignment_id
+      : [assignment_id];
+    const studentIds = Array.isArray(student_id) ? student_id : [student_id];
+
+    return { courseIds, assignmentIds, studentIds };
+  }, [queryParams]);
+
+  const [, navigate] = useLocation();
+  const updateFilters = useCallback(
+    ({ courseIds, assignmentIds, studentIds }: Partial<DashboardFilters>) => {
+      const newQueryParams = { ...queryParams };
```

Comment:
The base is `queryParams` and not `filters` to make sure unrelated query parameters are preserved.